test/multipart-disk.test.js

```diff
@@ -184,6 +185,104 @@ test('should throw on file limit error', async function (t) {
   }
 })
 
+test('should throw on file save error', async function (t) {
+  t.plan(2)
+
+  const fastify = Fastify()
+  t.tearDown(fastify.close.bind(fastify))
+
+  fastify.register(require('..'))
+
+  fastify.post('/', async function (req, reply) {
+    t.ok(req.isMultipart())
+
+    try {
+      await req.saveRequestFiles({ tmpdir: 'something' })
+      reply.code(200).send()
+    } catch (error) {
+      reply.code(500).send()
+    }
+  })
+
+  await fastify.listen(0)
+
+  // request
+  const form = new FormData()
+  const opts = {
+    protocol: 'http:',
+    hostname: 'localhost',
+    port: fastify.server.address().port,
+    path: '/',
+    headers: form.getHeaders(),
+    method: 'POST'
+  }
+  const req = http.request(opts)
+  const readStream = fs.createReadStream(filePath)
+  form.append('upload', readStream)
+
+  pump(form, req)
+
+  try {
+    const [res] = await once(req, 'response')
+    t.equal(res.statusCode, 500)
+    res.resume()
+    await once(res, 'end')
+  } catch (error) {
+    t.error(error, 'request')
+  }
+})
+
+test('should not throw on request files cleanup error', { skip: process.platform === 'win32' }, async function (t) {
```

Comment:
The reason for having this test skip the Windows platform is different.

I think the test is flaky on Windows. It failed on our windows CI with node 12.x  but it ran successfully on a Windows 10 laptop with node 12.x. (I can't prove that, you'll have to trust me :) )
My theory is that this could be similar to the Timing Caveat on windows described in [node-tap documentation](https://node-tap.org/docs/api/fixtures/) where removal of the `tempdir` is asynchronous. https://github.com/tapjs/node-tap/issues/677

In the test we are using rimraf which has a known issue that could be causing this: https://github.com/isaacs/rimraf/issues/72

Reply:
Makes sense, thanks for clarifying.

Reply:
you can remove rimraf. recursive folder removal is now part of Node.js. See https://nodejs.org/api/fs.html#fs_fs_rmdirsync_path_options and others.

Reply:
This module still supports Node 10 though

Reply:
Yes, recursive is not supported in Node 10.